common/src/main/java/com/wynntils/features/tooltips/ItemStatInfoFeature.java

```diff
@@ -90,37 +97,37 @@ public void onTooltipPre(ItemTooltipRenderEvent.Pre event) {
         WynnItem wynnItem = wynnItemOpt.get();
         if (brokenItems.contains(wynnItem)) return;
 
-        IdentifiableItemProperty itemInfo = getItemInfo(wynnItem);
-        if (itemInfo == null) return;
-
         try {
-            TooltipBuilder builder = wynnItem.getData()
-                    .getOrCalculate(
-                            WynnItemData.TOOLTIP_KEY,
-                            () -> Handlers.Tooltip.fromParsedItemStack(event.getItemStack(), itemInfo));
-            if (builder == null) return;
-
-            IdentificationDecorator decorator = identificationDecorations.get() ? new IdentificationDecorator() : null;
-            TooltipStyle currentIdentificationStyle = new TooltipStyle(
-                    identificationsOrdering.get(),
-                    groupIdentifications.get(),
-                    showBestValueLastAlways.get(),
-                    showStars.get());
-            LinkedList<Component> tooltips = new LinkedList<>(
-                    builder.getTooltipLines(Models.Character.getClassType(), currentIdentificationStyle, decorator));
-
-            // Update name depending on overall percentage; this needs to be done every rendering
-            // for rainbow/defective effects
-            if (overallPercentageInName.get() && itemInfo.hasOverallValue()) {
-                updateItemName(itemInfo, tooltips);
+            List<Component> tooltips = null;
+
+            Optional<IdentifiableItemProperty> identifiableItemPropertyOpt =
+                    Models.Item.asWynnItemProperty(event.getItemStack(), IdentifiableItemProperty.class);
+            if (identifiableItemPropertyOpt.isPresent()) {
+                tooltips =
+                        getIdentifiableItemTooltip(event.getItemStack(), wynnItem, identifiableItemPropertyOpt.get());
+            }
+
+            Optional<CraftedItemProperty> craftedItemPropertyOpt =
+                    Models.Item.asWynnItemProperty(event.getItemStack(), CraftedItemProperty.class);
+            if (craftedItemPropertyOpt.isPresent()) {
+                tooltips = getCraftedItemTooltip(event.getItemStack(), wynnItem, craftedItemPropertyOpt.get());
```

Comment:
This is kind of an "else" situation, right? If we've already gotten a tooltip, we can't get here.

Reply:
Yes, an item is either:
- identifiable: these are gear, tome, charms. items you identify/roll with the npc
- crafted items in essence also have stats/identifications, but they work differently, and we can't mix them with regular gear.

Since the relationship between crafted and normal gear is "so close yet so far", the tooltip building got into a bit of a messier situation. Overall, the code should be fine, code duplication is not big either.